src/pointer-analysis/value_set.cpp

```diff
@@ -195,7 +195,19 @@ void value_sett::output(
       else
       {
         result="<"+from_expr(ns, identifier, o)+", ";
-
+        if(o.id()==ID_dynamic_object)
+        {
+          dynamic_object_exprt::recencyt recency=
+            to_dynamic_object_expr(o).get_recency();
+          std::string recency_str="";
+          
+          if(recency==dynamic_object_exprt::recencyt::MOST_RECENT_ALLOCATION)
+            recency_str="most-recent-allocation";
+          else if(recency==dynamic_object_exprt::recencyt::ANY_ALLOCATION)
+            recency_str="any-allocation";
+          result+=recency_str+", ";
```

Comment:
This will put a comma in even if there isn't a recency. Is that what you intended? If not then you can combine this all into the cases above and not need recency_str at all.